core/executor/base_executor.go

```diff
@@ -57,17 +82,58 @@ func (e *BaseExecutor) GeneratePubData() error {
 	return nil
 }
 
-func (e *BaseExecutor) UpdateTrees() error {
-	return nil
-}
-
 func (e *BaseExecutor) GetExecutedTx() (*tx.Tx, error) {
 	e.tx.BlockHeight = e.bc.CurrentBlock().BlockHeight
-	e.tx.StateRoot = e.bc.StateDB().GetStateRoot()
-	e.tx.TxStatus = tx.StatusPending
+	e.tx.TxStatus = tx.StatusSuccess
+	e.tx.TxIndex = int64(len(e.bc.StateDB().Txs))
 	return e.tx, nil
 }
 
 func (e *BaseExecutor) GenerateTxDetails() ([]*tx.TxDetail, error) {
 	return nil, nil
 }
+
+func (e *BaseExecutor) MarkAccountAssetsDirty(accountIndex int64, assets []int64) {
+	if accountIndex < 0 {
+		return
+	}
+
+	_, ok := e.dirtyAccountsAndAssetsMap[accountIndex]
+	if !ok {
+		e.dirtyAccountsAndAssetsMap[accountIndex] = make(map[int64]bool, 0)
+	}
+
+	for _, assetIndex := range assets {
+		// Should never happen, but protect here.
+		if assetIndex < 0 {
+			continue
+		}
+		e.dirtyAccountsAndAssetsMap[accountIndex][assetIndex] = true
+	}
+}
+
+func (e *BaseExecutor) MarkLiquidityDirty(pairIndex int64) {
+	e.dirtyLiquidityMap[pairIndex] = true
+}
+
+func (e *BaseExecutor) MarkNftDirty(nftIndex int64) {
+	e.dirtyNftMap[nftIndex] = true
+}
+
+func (e *BaseExecutor) SyncDirtyToStateCache() {
```

Comment:
Looks like every exeutor will all `core/executor/base_executor.go`, I think we can simply it.

Reply:
fixed